### PR TITLE
feat(operator): add materialize. closes #132

### LINKED ISSUE #132: Add materialize

### PR DESCRIPTION
**Pull request by blesh**

### CODE REVIEW

File: src/Notification.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import Observer from './Observer';
+import Observable from './Observable';
+import noop from './util/noop';
+
+export default class Notification<T> {
+  hasValue: boolean;
+  
+  constructor(public kind: string, public value?: T, public exception?: any) {
+    this.hasValue = kind === 'N';
+    Object.freeze(this);
```
Comment (trxcllnt) [inline src/Notification.ts]: @blesh are there performance concerns around freezing every `Notification` upon initialization?

Comment (blesh) [inline src/Notification.ts]: @trxcllnt Good question! I thought about this when I was doing it, and I honestly don't know. The research I did led me to believe it was innocuous, but perhaps my sources were no good. I ran a JSPerf on such object creation, and it's about 70% slower.

... as such, I'll be removing that `freeze` block and avoiding it in the future. It was an ill-advised attempt at enforcing immutability.

**Comment by blesh** [comment]
Updated, @trxcllnt, please review. (All I did was remove the `freeze`)

**Comment by trxcllnt** [comment]
@blesh 👍

**Pull request merged**